sqlglot/dialects/tsql.py

```diff
@@ -539,3 +615,35 @@ def returning_sql(self, expression: exp.Returning) -> str:
             into = self.sql(expression, "into")
             into = self.seg(f"INTO {into}") if into else ""
             return f"{self.seg('OUTPUT')} {self.expressions(expression, flat=True)}{into}"
+
+        def transaction_sql(self, expression: exp.Transaction) -> str:
+            this = self.sql(expression, "this")
+            this = f" {this}" if this else ""
+            mark = expression.args.get("mark")
+            mark = f" WITH MARK {mark}" if mark else ""
+            return f"BEGIN TRANSACTION{this}{mark}"
+
+        def _durability_sql(self, expression) -> str:
+            durability = expression.args.get("durability")
+            durability_sql = ""
+            if durability is not None:
+                if durability:
+                    durability = "ON"
+                else:
+                    durability = "OFF"
+                durability_sql = f" WITH (DELAYED_DURABILITY = {durability})"
+            return durability_sql
```

Comment:
```suggestion
            if durability is not None:
                return f" WITH (DELAYED_DURABILITY = {'ON' if durability else 'OFF'})"
            return ""
```

Reply:
Let's just get rid of this whole method and move the suggested logic in `commit_sql` instead, since that's where `durability` is used.

Reply:
Done.